pkg/elfwriter/elfwriter_test.go

```diff
@@ -257,3 +258,88 @@ func TestWriter_WriteCompressedHeaders(t *testing.T) {
 
 	require.Equal(t, dIn, dOut)
 }
+
+func TestWriter_HasLinks(t *testing.T) {
+	inElf, err := elf.Open("testdata/libc.so.6")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		inElf.Close()
+	})
+
+	type fields struct {
+		FileHeader     *elf.FileHeader
+		Progs          []*elf.Prog
+		Sections       []*elf.Section
+		SectionHeaders []elf.SectionHeader
+	}
+	tests := []struct {
+		name                     string
+		fields                   fields
+		err                      error
+		expectedNumberOfSections int
+		isSymbolizable           bool
+	}{
+		{
+			name: "only keep file header",
+			fields: fields{
+				FileHeader: &inElf.FileHeader,
+			},
+		},
+		{
+			name: "only keep program header",
+			fields: fields{
+				FileHeader: &inElf.FileHeader,
+				Progs:      inElf.Progs,
+			},
+		},
+		{
+			name: "keep all sections and segments",
+			fields: fields{
+				FileHeader: &inElf.FileHeader,
+				Progs:      inElf.Progs,
+				Sections:   inElf.Sections,
+			},
+			expectedNumberOfSections: len(inElf.Sections),
+			isSymbolizable:           true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output, err := os.Create("test-output-links-here")
```

Comment:
Sorry for the late comment, but should we remove this file and/or add it to the `.gitignore`? It's gets generated when running the tests, but it's never cleaned up

Reply:
Yes, sorry about it. That was for some local debugging. Here you go: https://github.com/parca-dev/parca-agent/pull/1005